core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -331,30 +335,36 @@ public void removeBlocks(List<Long> blockIds, boolean delete) {
         continue;
       }
       HashSet<Long> workerIds = new HashSet<>();
-      synchronized (block) {
-        // Technically, 'block' should be confirmed to still be in the data structure. A
-        // concurrent removeBlock call can remove it. However, we are intentionally ignoring this
-        // race, since deleting the same block again is a noop.
-        workerIds.addAll(block.getWorkers());
-        // Two cases here:
-        // 1) For delete: delete the block metadata.
-        // 2) For free: keep the block metadata. mLostBlocks will be changed in
-        // processWorkerRemovedBlocks
-        if (delete) {
-          // Make sure blockId is removed from mLostBlocks when the block metadata is deleted.
-          // Otherwise blockId in mLostBlock can be dangling index if the metadata is gone.
-          mLostBlocks.remove(blockId);
-          mBlocks.remove(blockId);
+      try (JournalContext journalContext = createJournalContext()) {
+        synchronized (block) {
+          // Technically, 'block' should be confirmed to still be in the data structure. A
+          // concurrent removeBlock call can remove it. However, we are intentionally ignoring this
+          // race, since deleting the same block again is a noop.
+          workerIds.addAll(block.getWorkers());
+          // Two cases here:
+          // 1) For delete: delete the block metadata.
+          // 2) For free: keep the block metadata. mLostBlocks will be changed in
+          // processWorkerRemovedBlocks
+          if (delete) {
+            // Make sure blockId is removed from mLostBlocks when the block metadata is deleted.
+            // Otherwise blockId in mLostBlock can be dangling index if the metadata is gone.
+            mLostBlocks.remove(blockId);
+            mBlocks.remove(blockId);
+            JournalEntry entry = JournalEntry.newBuilder()
+                .setDeleteBlock(DeleteBlockEntry.newBuilder().setBlockId(blockId)).build();
+            appendJournalEntry(entry, journalContext);
```

Comment:
Only journal this if the block was removed from the data structure? Otherwise since we call removeBlocks on file system master journal replay, I think we will generate a significant amount of unnecessary logs